src/ruby.rs

```diff
@@ -51,100 +51,890 @@ extern {
+    ///
+    /// ## Exceptions
+    ///
+    /// * [`rb_eTypeError`](static.rb_eTypeError.html)
+    ///     * if `symbol` is not a `rb_cSymbol`
+    ///
+    /// # Miscellaneous
+    ///
+    /// [`RB_SYM2ID`](https://github.com/ruby/ruby/blob/v2_5_1/include/ruby/ruby.h#L381) and
+    /// [`SYM2ID`](https://github.com/ruby/ruby/blob/v2_5_1/include/ruby/ruby.h#L386) are currently
+    /// aliases for this.
+    ///
+    /// # Defined In
+    ///
+    /// * **2.5:**
+    ///     [ruby.h](https://github.com/ruby/ruby/blob/v2_5_1/include/ruby/ruby.h#L375) |
+    ///     [symbol.c](https://github.com/ruby/ruby/blob/v2_5_1/symbol.c#L697-L722)
     pub fn rb_sym2id(symbol: VALUE) -> ID;
+
+    /// Convert an [`ID`](struct.ID.html) to a [`rb_cSymbol`](static.rb_cSymbol.html)
+    ///
+    /// * Returns an instance of [`rb_cSymbol`](static.rb_cSymbol.html)
+    ///
+    /// # Safety
+    ///
+    /// * Behavior is undefined if the `ID` is not valid.
+    ///
+    /// # Defined In
+    ///
+    /// * **2.5:**
+    ///     [ruby.h](https://github.com/ruby/ruby/blob/v2_5_1/include/ruby/ruby.h#L376) |
+    ///     [symbol.c](https://github.com/ruby/ruby/blob/v2_5_1/symbol.c#L725-L730)
     pub fn rb_id2sym(id: ID) -> VALUE;
 
+    /// Define a new class
+    ///
+    /// NOTE: If the class is already defined and the superclass is the same
+    /// as specified, it will return the already defined class.
+    ///
+    /// * `name` - nul-terminated C string, treated as ASCII encoded
+    /// * `superclass`: [`rb_cClass`](static.rb_cClass.html)
+    /// * Returns a [`rb_cClass`](static.rb_cClass.html)
+    ///
+    /// # Safety
+    ///
+    /// ## Exceptions
+    ///
+    /// * [`rb_eTypeError`](static.rb_eTypeError.html)
+    ///     * if the constant name is already taken, but the constant is not a C class
+    ///     * if the class is already defined and the superclass does not match
+    /// * [`rb_eArgError`](static.rb_eArgError.html)
+    ///     * if the superclass is null
+    ///
+    /// # Defined In
+    ///
+    /// * **2.3:** [ruby.h](https://github.com/ruby/ruby/blob/v2_3_7/include/ruby/ruby.h#L1656)
+    /// * **2.4:** [ruby.h](https://github.com/ruby/ruby/blob/v2_4_4/include/ruby/ruby.h#L1676)
+    /// * **2.5:**
+    ///     [ruby.h](https://github.com/ruby/ruby/blob/v2_5_1/include/ruby/ruby.h#L1678) |
+    ///     [class.c](https://github.com/ruby/ruby/blob/v2_5_1/class.c#L645-L673) |
+    ///     documentation: [usage](https://ruby-doc.org/core-2.5.1/doc/extension_rdoc.html#label-Class+and+Module+Definition),
+    ///                     [spec](https://ruby-doc.org/core-2.5.1/doc/extension_rdoc.html#label-Defining+Classes+and+Modules)
+    /// * **2.6:** [ruby.h](https://github.com/ruby/ruby/blob/v2_6_0_preview2/include/ruby/ruby.h#L1695)
     pub fn rb_define_class(name: *const c_char, superclass: VALUE) -> VALUE;
 
+    /// Defines a public method on a class
+    ///
+    /// * `class`: a [`rb_cClass`](static.rb_cClass.html)
```

Comment:
inconsistent use of `:` vs `-` (in multiple places around here)

Reply:
I was using colon for type specifications and dash for description, but I realize that isn’t necessarily obvious, so probably unhelpful.